test/audits/vulnerabilities.test.js

```diff
@@ -537,19 +556,387 @@ describe('Vulnerabilities Handler Integration Tests', () => {
+
+      const result = await opportunityAndSuggestionsStep(context);
+      expect(result).to.deep.equal({ status: "complete" });
+
+      // Verify SQS message to mystique was sent
+      expect(context.sqs.sendMessage).to.have.been.calledOnce;
+
+      // Verify the message structure
+      const messageCall = context.sqs.sendMessage.getCall(0);
+      const message = messageCall.args[1];
 
-      const auditData = createAuditData();
+      expect(context.sqs.sendMessage).to.have.been.calledOnce;
+      expect(message).to.have.property("type", "codefix:security-vulnerabilities");
+      expect(message).to.have.property("siteId", context.site.getId());
+      expect(message).to.have.property("auditId", "test-audit-id");
+      expect(message).to.have.property("deliveryType", "aem_cs");
+      expect(message.data).to.have.property("importResults", codeData);
+      expect(message.data).to.have.property("suggestions");
+    });
+
+    it("should handle configuration lookup failure gracefully", async () => {
+      context.dataAccess.Configuration.findLatest.rejects(new Error("Database connection failed"));
+
+      context.audit = {
+        getAuditResult: () => ({
+          vulnerabilityReport: VULNERABILITY_REPORT_WITH_VULNERABILITIES,
+          success: true,
+        }),
+        getId: () => "test-audit-id",
+      };
 
       // This should throw an error since the handler doesn't
       // handle config lookup failures gracefully
-      await expect(opportunityAndSuggestionsStep('https://example.com', auditData, context, site))
-        .to.be.rejectedWith('Database connection failed');
+      await expect(opportunityAndSuggestionsStep(context)).to.be.rejectedWith("Database connection failed");
+    });
+  });
+
+  describe("extractCodeBucket", () => {
+    it("should return code bucket data when audit succeeds", async () => {
+      // Setup successful audit
+      setupSuccessfulImsAuth();
+      setupSuccessfulVulnerabilityApi();
+
+      const { extractCodeBucket } = await import("../../src/vulnerabilities/handler.js");
+
+      const result = await extractCodeBucket(context);
+
+      expect(result).to.have.property("type", "code");
+      expect(result).to.have.property("siteId", "a1b2c3d4-e5f6-7890-abcd-ef1234567890");
+      expect(result).to.have.property("auditResult");
+      expect(result).to.have.property("fullAuditRef", "https://example.com");
+      expect(result.auditResult).to.have.property("success", true);
+    });
+
+    it("should throw error when audit fails", async () => {
+      // Setup failed audit
+      context.site.getDeliveryType = () => "other";
+
+      const { extractCodeBucket } = await import("../../src/vulnerabilities/handler.js");
+
+      try {
+        await extractCodeBucket(context);
+      } catch (error) {
+        expect(error.message).to.equal("Audit failed, skipping call to import worker");
+      }
     });
   });
 });
+
+// NOTE: fully AI generated tests
```

Comment:
Please clean up the tests. Honestly, AI tests are fine, but lets still make sure to not bloat the test set. This is just unecessary and has no purpose